src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -827,6 +818,16 @@ fn configure_llvm(builder: &Builder<'_>, target: TargetSelection, cfg: &mut cmak
         }
     }
 
+    // Libraries for ELF compression.
+    if !target.is_windows() {
+        cfg.define("LLVM_ENABLE_ZLIB", "ON");
+        cfg.define("LLVM_ENABLE_ZSTD", "ON");
+        cfg.define("LLVM_USE_STATIC_ZSTD", "TRUE");
+    } else {
+        cfg.define("LLVM_ENABLE_ZLIB", "OFF");
+        cfg.define("LLVM_ENABLE_ZSTD", "OFF");
+    }
+
```

Comment:
lld has its own separate usage of zstd outside of libllvm.so. e.g. https://github.com/llvm/llvm-project/blob/edf4e02906bd0c57087a44e5c4890ffc453667b2/lld/ELF/CMakeLists.txt#L9 https://github.com/llvm/llvm-project/blob/edf4e02906bd0c57087a44e5c4890ffc453667b2/lld/ELF/OutputSections.cpp#L387

Because these options aren't currently applied in the Step impl for Lld, right now lld is just built with the defaults from LLVM. If you do a complete build of Rust/LLVM today on a machine with libzstd-dev/equivalent installed the resulting rust-lld will dynamically link to libzstd.so, because LLVM defaults LLVM_ENABLE_ZSTD to ON and LLVM_USE_STATIC_ZSTD to FALSE.